ext/hal/nxp/imx/drivers/clock_freq.c

```diff
@@ -0,0 +1,243 @@
+/*
```

Comment:
The files `clock_freq.h/c` are coming from examples folder in `FreeRTOS_BSP_1.0.1_iMX7D` and the content of those files is different in `FreeRTOS_BSP_1.0.1_iMX6SX`. It should be renamed or moved, maybe into `ext/hal/nxp/imx/devices/MCIMX7D/` folder?

Reply:
Do you think that the functions in this module can be the same for i.MX7 and i.MX6SX?
If so, it doesn't make sense to move those files to `ext/hal/nxp/imx/devices/MCIMX7D/`

Reply:
I haven't looked at the two set of files in much depth yet, but theirs diff is quite large.
The purpose of the files (getting frequency of SoC's particular subsystems) is similar to what can be found for example in `ext\hal\nxp\mcux\devices\MIMXRT1052\fsl_clock.c`, which is device specific.
So I am afraid there will have to be versions of clock_freq.h/c for each SoC based on FreeRTOS BSP (KSDK 1.x).

Reply:
Ok. I'll move them to `ext/hal/nxp/imx/devices/MCIMX7D/`.

Reply:
It also looks like the proper Zephyr way is to create abstraction of the device specific clock code (in `drivers/clock_control`) and use it instead of accessing the external API directly in driver shims. We may consider adding clock_control module for iMX6SX/7D in future.

Reply:
Agreed. My idea is to have this basic support upstreamed and than implement other drivers (GPIO, I2C etc) and frameworks (clock_control, pinmux etc).